dotCMS/src/main/java/com/dotmarketing/cms/urlmap/filters/URLMapFilter.java

```diff
@@ -232,21 +236,32 @@ public void doFilter(ServletRequest req, ServletResponse res, FilterChain chain)
                         }
 
                         cons = conAPI.searchIndex(query.toString(), 2, 0,
-                                (hostIsRequired ? "conhost, modDate" : "modDate"), user, true);
+                                (hostIsRequired ? "conhost, modDate" : "modDate"), wuserAPI.getSystemUser(), true);
                         int idx = 0;
                         if (checkIndex && cons.size() == 2) {
                             // prefer session setting
-                            Contentlet second = conAPI.find(cons.get(1).getInode(), user, true);
+                            Contentlet second = conAPI.find(cons.get(1).getInode(), wuserAPI.getSystemUser(), true);
                             if (second.getLanguageId() == sessionLang) {
                                 idx = 1;
                             }
                         }
                         ContentletSearch c = cons.get(idx);
+                        Contentlet contentlet = conAPI
+                                .find(c.getInode(), wuserAPI.getSystemUser(), true);
+
                         if (optSession != null) {
                             optSession.setAttribute(com.dotmarketing.util.WebKeys.HTMLPAGE_LANGUAGE,
-                                    String.valueOf(
-                                            conAPI.find(c.getInode(), user, true).getLanguageId()));
+                                    String.valueOf(contentlet.getLanguageId()));
+                        }
+
+                        //Verify and handle the case for unauthorized access of this contentlet
+                        Boolean unauthorized = CMSUrlUtil.getInstance()
+                                .isUnauthorizedAndHandleError(contentlet, uri, user, request,
+                                        response);
+                        if (unauthorized) {
```

Comment:
![CRITICAL](https://sonarsource.github.io/sonar-github/severity-critical.png 'Severity: CRITICAL') Refactor this code to not nest more than 4 if/for/while/switch/try statements. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS134)